### ISSUE TITLE
[best-practices-programming] python: quit() vs sys.exit(1)

### ISSUE THREAD

**Issue by jgvictores** [roboticslab-uc3m/developer-manual#43]
[best-practices-programming] python: quit() vs sys.exit(1)

Just putting here to document. Not a serious issue, just in the light of https://github.com/roboticslab-uc3m/yarp-devices/commit/3c1f7f367dd637523dfe65b3335a516d5ca9d4ef where both were being used. Could not find similar issue in https://github.com/roboticslab-uc3m/questions-and-answers.

**Comment by PeterBowman** [comment]
See also comment to commit https://github.com/roboticslab-uc3m/kinematics-dynamics/commit/127a768baa478bb7194762435c64220f2022bd19 and https://stackoverflow.com/a/19747562.

**Comment by jgvictores** [comment]
Yup, I agree with my past self that `raise SystemExit` saves you from `import sys`. A few notes:

- Both `exit(1)` and `raise SystemExit(1)` are valid syntax.
- There is a small goodie in `exit`: "`exit("Here is why the program failed")`, which prints the error message and exits with return status 1." (via `Jasha` at mentioned SO answer).
